selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/CheTestWorkspaceServiceClientFactory.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.core.client;
+
+import com.google.inject.assistedinject.Assisted;
+import org.eclipse.che.selenium.core.user.TestUser;
+
+public interface CheTestWorkspaceServiceClientFactory extends TestWorkspaceServiceClientFactory {
```

Comment:
It ought to be `TestWorkspaceServiceClientFactory` which produces `TestWorkspaceServiceClient` instance.

Reply:
I see but how should we then differentiate to include RhCheTestWorkspaceServiceClient or CheTestWorkspaceServiceClient ?

Reply:
Guice will substitute proper concrete class automatically depending on what is bound in the module class.

Reply:
So, we doesn't need this class `CheTestWorkspaceServiceClientFactory` if we have already had `TestWorkspaceServiceClientFactory`.